draft-xu-risav.md

```diff
@@ -240,7 +240,17 @@ Conversely, if any AS no longer publishes a `RISAVAnnouncement`, other ASes MUST
 
 > OPEN QUESTION: Does IKEv2 have an authenticated permanent rejection option that would help here?
 
+## Green Channel
 
+When ACS is downing, the SA generation may also fail before ACS reboots. ASBR could also suffer from such as misconfiguration, power failure, etc.
+
+Thus it is the first thing to guarantee the traffic flow when ACS or ASBR is stopped. But the SA generation may be wrong at that time, which means the key generation is not consistent between two communication ASes, it is hard to recover the running system from such a situation.
+
+It needs a mechanism to restore the control flow first, specifically the communication between public IP pairs. RISAV provides a green channel for public IP pairs.
+
+It is a switch that when the green channel is enabled, the packets with the source address and destination address which are all public IPs MUST NOT be inspected by the RISAV headers. Thus the control packet for restoring RISAV could proceed without hindrance.
+
+However, when the green channel is disabled, the packet with such public IPs at their source address or destination address MUST be inspected as ordinary traffic.
```

Comment:
Consider this alternative formulation:

```suggestion
In the event of a misconfiguration or loss of state, it is possible that a negotiated SA could become nonfunctional before its expiration time.  For example, if one AS is forced to reset its ACS and ASBRs, it may lose the private keys for all active RISAV SAs.  If RISAV were applied to the IKEv2 traffic used for bootstrapping, the participating ASes would be unable to communicate until these broken SAs expire, likely after multiple hours or days.

To ensure that RISAV participants can rapidly recover from this error state, RISAV places bootstrapping traffic in a "green tunnel" that is exempt from RISAV's protections.  This "tunnel" is defined by two requirements:

* RISAV senders MUST NOT add RISAV protection to packets to or from any announced contact IP
* RISAV recipients MUST NOT enforce RISAV validation on packets sent to or from any announced contact IP.

Although the green tunnel denies RISAV protection to the ACS, the additional mitigations described in {data-plane} ensure that the ACS is nonetheless strongly protected against address-spoofing and DDoS attacks.
```

Apart from phrasing, the only difference here is that I have omitted the discussion of disabling the green tunnel.  I think it's a very good idea, so we don't need a way to disable it.

Reply:
Thank you. It reads more smoothly now.

But if all the green channels are open, the security of contact IP or ACS would be vulnerable. It seems the following description in section {data-plane} can not prevent such a spoofing scenario that Attacker `A` uses the contact IP of `S` as its source address and sends the packet to `D`.

In this case, if ASBR of `D` does not inspect but removes the RISAV header, this packet is sent to the ACS of `D`. I don't know if it is possible to prevent the end hosts, not ACS, in one participating AS from sending such packets. If the operator could close the green channel manually, the above scenario may disappear as the ASBR of `D` would inspect the RISAV header of incoming packets.

So I think it is better to close the green channel if the communication is stable. There may be some subtle details that bother us, for instance, it could not be able to enable/disable the green channel automatically and smartly.

> To avoid DoS attacks, participating ASes MUST drop any outgoing packet to the contact IP of another AS. Only the AS operator's systems (i.e. the ACS and ASBRs) are permitted to send packets to the contact IPs of other ASes. ASBRs MAY drop inbound packets to the contact IP from non-participating ASes.

Reply:
I see your point: an attacker in an arbitrary location could spoof the source IP of the contact IP of one AS in order to attack another AS.  I think this may be tolerable.  After all, the ACS is part of the RISAV system, which is expected to absorb and reject these attacks.

In the case of the ACS, I think the IKEv2 COOKIE system is sufficient to reject off-path attacks (at low cost), and the RPKI authentication rejects on-path attacks (at a higher cost).

Providing a way to disable the "green tunnel" would reduce the cost of rejecting on-path attacks, but it seems to create a high risk of outages.  As you noted, it also adds complexity to support enabling and disabling the "green tunnel" between a pair of ASes.

My preference would be to add some text mentioning the COOKIE mitigation and the remaining attacks, and not try to add more mitigations for this.